mesonbuild/options.py

```diff
@@ -250,19 +250,23 @@ def evolve(self,
                          subproject if subproject != _BAD_VALUE else self.subproject, # None is a valid value so it can'the default value in method declaration.
                          machine if machine is not None else self.machine)
 
-    def as_root(self) -> 'OptionKey':
-        """Convenience method for key.evolve(subproject='')."""
-        if self.subproject is None or self.subproject == '':
+    def as_root(self) -> OptionKey:
+        """Convenience method for key.evolve(subproject='') if subproject is not global."""
+        if not self.subproject:
             return self
-        return self.evolve(subproject='')
+        return OptionKey(self.name, subproject='', machine=self.machine)
+
+    def as_toplevel(self) -> OptionKey:
+        """Convenience method for key.evolve(subproject='')."""
+        return OptionKey(self.name, subproject='', machine=self.machine)
```

Comment:
I don't see any justification for why self.evolve() was replaced here?

Reply:
because self.evolve requires a function call and 3 comparisons more.

Reply:
I'm surprised that shows up in the profile honestly.

Since it does, you'll need to document that in the comment or someone like me is going to come by and say "why doesn't this just use self.evolve? that's a lot less typing"

Reply:
You're right. It doesn't seem to make a big difference. Let use evolve then!